libsession/src/main/java/org/session/libsession/messaging/jobs/RetrieveProfileAvatarJob.kt

```diff
@@ -0,0 +1,95 @@
+package org.session.libsession.messaging.jobs
+
+import android.text.TextUtils
+import org.session.libsession.avatars.AvatarHelper
+import org.session.libsession.messaging.MessagingModuleConfiguration
+import org.session.libsession.messaging.utilities.Data
+import org.session.libsession.utilities.DownloadUtilities.downloadFile
+import org.session.libsession.utilities.TextSecurePreferences.Companion.setProfileAvatarId
+import org.session.libsession.utilities.Util.copy
+import org.session.libsession.utilities.Util.equals
+import org.session.libsession.utilities.Address
+import org.session.libsession.utilities.recipients.Recipient
+import org.session.libsignal.streams.ProfileCipherInputStream
+import org.session.libsignal.utilities.Log
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileOutputStream
+import java.io.InputStream
+import java.security.SecureRandom
+
+class RetrieveProfileAvatarJob(val profileAvatar: String, val recipientAddress: Address): Job {
+    override var delegate: JobDelegate? = null
+    override var id: String? = null
+    override var failureCount: Int = 0
+    override val maxFailureCount: Int = 0
+
+    companion object {
+        val TAG = RetrieveProfileAvatarJob::class.simpleName
+        val KEY: String = "RetrieveProfileAvatarJob"
+
+        // Keys used for database storage
+        private val PROFILE_AVATAR_KEY = "profileAvatar"
+        private val RECEIPIENT_ADDRESS_KEY = "recipient"
+    }
+
+    override fun execute(dispatcherName: String) {
+        val context = MessagingModuleConfiguration.shared.context
+        val storage = MessagingModuleConfiguration.shared.storage
+        val recipient = Recipient.from(context, recipientAddress, true)
+        val profileKey = recipient.resolve().profileKey
+
+        if (profileKey == null || (profileKey.size != 32 && profileKey.size != 16)) {
+            Log.w(TAG, "Recipient profile key is gone!")
+            return
+        }
+
+        if (AvatarHelper.avatarFileExists(context, recipient.resolve().address) && equals(profileAvatar, recipient.resolve().profileAvatar)) {
+            Log.w(TAG, "Already retrieved profile avatar: $profileAvatar")
+            return
+        }
+
+        if (TextUtils.isEmpty(profileAvatar)) {
```

Comment:
now that this is kotlin we probably get some nice `profileAvatar.isEmpty()` function we can call, but since we are doing this check we might want to make `profileAvatar` a nullable string, then do a null or empty check as well, so we don't have to pass in an empty string to clear it (that might cause issues) :thinking:

Reply:
Have updated the job to take a nullable `profileAvatar`